pkg/kubectl/resource/result.go

```diff
@@ -160,12 +160,27 @@ func (r *Result) Object() (runtime.Object, error) {
 	if len(versions) == 1 {
 		version = versions.List()[0]
 	}
-	return &api.List{
+
+	return toV1List(objects, version), err
```

Comment:
This is still resulting in a v1.List containing internal objects. Is there a reason not to use VersionedObject above when building the `objects` array?

Reply:
What is the impact of the change you suggest?  It seems much more invasive than this PR.  How does it impact `get` and `rollingupdate` which are currently using the internal version?

Reply:
@liggitt FWIW, what you are suggesting may be perfectly safe, I just don't know enough about the area.

Reply:
> What is the impact of the change you suggest? 

I'm not sure yet

> I read @smarterclayton's response as doing it this way is fine, and a step in the right direction.

I read "v1.List should work with external objects (internal.List can also have external objects)" to mean that we shouldn't put internal objects in a v1.List, but it's possible I misunderstood.

I *think* wrapping internal objects in an external list is incorrect, and am concerned it is introducing bugs in reserialization. I pulled down this branch to hit the obvious cases and couldn't find any issues, so I'm tentatively ok with proceeding to unblock consumers of the resource package, but I think it's worthwhile to switch to versioned in an immediate follow up and chase the ripples through the kubectl commands that require internal.

Reply:
@liggit Thx.  I will follow up this week.  You may be right about @smarterclayton's comment, reading it a second time I am less certain.